src/Symfony/Component/Console/Command/CompleteCommand.php

```diff
@@ -77,16 +80,16 @@ protected function initialize(InputInterface $input, OutputInterface $output)
     protected function execute(InputInterface $input, OutputInterface $output): int
     {
         try {
-            // uncomment when a bugfix or BC break has been introduced in the shell completion scripts
-            //$version = $input->getOption('symfony');
-            //if ($version && version_compare($version, 'x.y', '>=')) {
-            //    $message = sprintf('Completion script version is not supported ("%s" given, ">=x.y" required).', $version);
-            //    $this->log($message);
+            // "symfony" must be kept for compat with the shell scripts generated by Symfony Console 5.4 - 6.1
+            $version = $input->getOption('symfony') ? '1' : $input->getOption('api-version');
+            if ($version && version_compare($version, self::COMPLETION_API_VERSION, '>=')) {
+                $message = sprintf('Completion script version is not supported ("%s" given, ">=x.y" required).', $version);
```

Comment:
The error message should be dynamic.
```suggestion
                $message = sprintf('Completion script version is not supported ("%s" given, ">=%s" required).', $version, self::COMPLETION_API_VERSION);
```